shared/contracts/BaseTemplate.sol

```diff
@@ -56,155 +62,178 @@ contract BaseTemplate is APMNamehash, IsContract {
         dao = daoFactory.newDAO(this);
         emit DeployDao(address(dao));
         acl = ACL(dao.acl());
-        _createPermission(acl, dao, dao.APP_MANAGER_ROLE());
+        _createPermissionForTemplate(acl, dao, dao.APP_MANAGER_ROLE());
     }
 
     /* ACL */
 
-    function _createPermission(ACL acl, address app, bytes32 permission) internal {
-        acl.createPermission(address(this), app, permission, address(this));
+    function _createPermissions(ACL _acl, address[] _grantees, address _app, bytes32 _permission, address _manager) internal {
```

Comment:
One thing I'm not sure about is on the argument order for these base functions.

It's true that the ACL has its arguments in order of `entity`, `app`, and `role`, but that's because it's in the context of the ACL (and not a template).

I wonder if it would make more sense (or maybe it's more confusing?) if we re-ordered the arguments in the templates as `acl`, `app`, `role`, `grantee/s`, `manager`, so that `acl`, `app`, and `role` are consistently in the same order at the start.

Reply:
Agreed with @bingen, I like the idea, but I prefer consistency. WDYT about replicating that in the `ACL` @sohkai ?

Reply:
We should avoid changes to the ACL's API for the foreseeable future, until there's a _really_ good reason to change it, so let's keep the order as is (`entity`, `app`, `role`, `manager`).